src/rules/consistent_generic_constructors.rs

```diff
@@ -0,0 +1,677 @@
+                        node.child_by_field_name("type").map(|type_| type_.first_non_comment_named_child(SupportedLanguage::Javascript)),
+                        node.child_by_field_name("value")
+                    ),
+                    _ => unreachable!()
+                };
+                let Some(rhs) = rhs.filter(|&rhs| {
+                    rhs.kind() == NewExpression &&
+                        rhs.field("constructor").kind() == Identifier
+                }) else {
+                    return;
+                };
+
+                match self.mode {
+                    Options::TypeAnnotation => {
+                        if lhs.is_some() {
+                            return;
+                        }
+                        let Some(type_arguments) = rhs.child_by_field_name("type_arguments") else {
+                            return;
+                        };
+                        let callee = rhs.field("constructor");
+                        let type_annotation = format!(
+                            "{}{}",
+                            callee.text(context),
+                            type_arguments.text(context),
+                        );
+                        context.report(violation! {
+                            node => node,
+                            message_id => "prefer_type_annotation",
+                            fix => |fixer| {
+                                let id_to_attach_annotation = match node.kind() {
+                                    PublicFieldDefinition => node.field("name"),
+                                    _ => lhs_name
+                                };
+                                fixer.remove(type_arguments);
+                                fixer.insert_text_after(
+                                    id_to_attach_annotation,
+                                    format!(": {type_annotation}")
+                                );
+                            }
+                        });
+                    }
+                    Options::Constructor => {
+                        if rhs.child_by_field_name("type_arguments").is_some() {
+                            return;
+                        }
+                        let Some(lhs_type_arguments) = lhs.filter(|&lhs| {
+                            lhs.kind() == GenericType &&
+                                lhs.field("name").thrush(|lhs_name| {
+                                    lhs_name.kind() == TypeIdentifier &&
+                                        lhs_name.text(context) == rhs.field("constructor").text(context)
+                                })
+                        }).map(|lhs| lhs.field("type_arguments")) else {
+                            return;
+                        };
+                        let lhs = lhs.unwrap();
+                        let has_parens = rhs.child_by_field_name("arguments").is_some();
+                        let mut extra_comments: IndexSet<Node<'a>> = context
```

Comment:
Use `IndexSet` (from `indexmap`) to ensure a consistent ordering (so that comments will get spit back out in the order you read them basically)

Which maybe implies that the more appropriate data structure here is a list rather than a set? But whatever the typescript-eslint version uses a set